crates/resolver/src/name_server/connection_provider.rs

```diff
@@ -379,17 +379,22 @@ impl Stream for ConnectionResponse {
 #[allow(unreachable_pub)]
 pub mod tokio_runtime {
     use super::*;
+    use std::sync::{Arc, Mutex};
     use tokio::net::UdpSocket as TokioUdpSocket;
+    use tokio::task::JoinSet;
 
     /// A handle to the Tokio runtime
-    #[derive(Clone, Copy)]
-    pub struct TokioHandle;
+    #[derive(Clone, Default)]
```

Comment:
Is there a concern about creating new joinsets accidentally? What's the implication of creating a new one in a context where others exist? Could one joinset cause a thread to block while potentially wanting on a separate set of futures in a different joinset?

If any of these are concerns, we should probably warn that it's desirable to have a single `TokioHandle` in a context.

Reply:
A JoinSet simply scopes all tasks spawned on it (within a running tokio context, of which there's one per program), and allows awaiting/aborting/dropping that set of tasks. It's normal and expected to have more than one. They shouldn't interfere, as long as there's no blocking in there already.

I'm not sure the rustdoc is accurate here anymore, as this isn't a handle to the runtime anymore. But more importantly, per your below comment, is there any reason to keep this public?

Reply:
👍